### PR TITLE
Fix scenario move_dir_and_update_subfile_remotely

### PR DESCRIPTION
**Pull request by aenario**
This is PR is split from #1250 which was unreadable.

### CODE REVIEW

File: core/metadata.js [hunk 1]
```diff
@@ -256,6 +257,12 @@ function isUpToDate (side /*: SideName */, doc /*: Metadata */) {
   return currentRev === lastRev
 }
 
+function isAtLeastUpToDate (side /*: SideName */, doc /*: Metadata */) {
+  let currentRev = doc.sides[side] || 0
+  let lastRev = extractRevNumber(doc)
+  return currentRev >= lastRev
```
Comment (sebn) [inline core/metadata.js]: When I change `isUpToDate()` to `return currentRev >= lastRev` on master, tests (unit/integration/scenarios) pass on my laptop. I pushed #1276 to see what happens on CI.

Actually, all the conflict integration tests except one pass on master without any change.

The only test that fails (`local applied first -> conflict`) is effectively fixed by the new dissociation code in Merge, but it doesn't need the `isUpToDate()` change either.

File: core/sync.js [hunk 2]
```diff
@@ -297,6 +297,9 @@ class Sync {
       } else {
         await this.doMove(side, doc, from)
       }
+      if (!sameBinary(from, doc)) {
+        await side.overwriteFileAsync(doc, from)
+      }
```
Comment (sebn) [inline core/sync.js]: It doesn't seem to fix the scenario on CI:
https://travis-ci.org/cozy-labs/cozy-desktop/jobs/454611472#L377

I tried every combo of `isUpToDate` / `isAtLeastUpToDate`, but none of them fix it either.
So I assume something else in #1250 was making it work.

File: core/metadata.js (hunk 1 above)
Comment (aenario) [reply core/metadata.js]: :thinking: Either I got confused, or it has more interactions with the #1274 than i thought. Let's merge that one first and then we will recheck.

File: core/sync.js (hunk 2 above)
Comment (aenario) [reply core/sync.js]: Looks like it, let's merge #1274 and then we will see.

File: test/integration/conflict_resolution.js [hunk 3]
```diff
@@ -52,6 +55,101 @@ describe('Conflict resolution', () => {
...
+        md5sum: await helpers.local.syncDir.checksum('concurrent-edited')
+      }))
+    }
+
+    const expectedTree = [
+      'concurrent-edited',
+      'concurrent-edited-conflict-...'
+    ]
+
+    it('local applied first -> conflict', async () => {
```
Comment (sebn) [inline test/integration/conflict_resolution.js]: Couldn't *applied* be confused with `Sync#apply()`?
What about *merged*?

File: test/integration/conflict_resolution.js [hunk 4]
```diff
@@ -52,6 +55,101 @@ describe('Conflict resolution', () => {
...
+      await simulateLocalUpdateEvent()
```
Comment (sebn) [inline test/integration/conflict_resolution.js]: Idem, *simulateLocalUpdateMerge*?

File: core/merge.js [hunk 5]
```diff
@@ -216,8 +217,16 @@ class Merge {
...
-      } else if (!isUpToDate(side, file)) {
-        return this.resolveConflictAsync(side, doc, file)
+      } else if (!isAtLeastUpToDate(side, file)) {
+        await this.resolveConflictAsync(side, doc, file)
+        if (side === 'remote') {
+          // we just renamed the remote file as a conflict
+          // the old file should be dissociated from the remote
+          delete file.remote
+          if (file.sides) delete file.sides.remote
+          await this.pouch.put(file)
```
Comment (sebn) [inline core/merge.js]: Should we be doing this in other Merge methods too?

**Comment by sebn** [comment]
Should we close this one in favor of #1285?

**Comment by aenario** [comment]
This have one more commit than #1285, let's merge that one first and then rebase this one.

**Comment by aenario** [comment]
Rebased on master after merging #1285 , let's wait for travis

File: core/sync.js (hunk 2 above)
Comment (aenario) [reply core/sync.js]: I opened a PR with only this fix to see if it's sufficient in https://github.com/cozy-labs/cozy-desktop/pull/1290/files

**Pull request merged**